client-tools/Import/SenderThread.cpp

```diff
@@ -161,6 +161,8 @@ void SenderThread::handleResult(httpclient::SimpleHttpResult* result) {
     return;
   }
 
+  std::unique_lock guardError{_condition.mutex};
```

Comment:
This probably works, but it _may be_ a pessimization to hold the mutex while parsing the response body.
`getBodyVelocyPack()` does the following:
```
std::shared_ptr<VPackBuilder> SimpleHttpResult::getBodyVelocyPack() const {
  std::string_view data(_resultBody.c_str(), _resultBody.length());
  VPackParser parser(&VelocyPackHelper::looseRequestValidationOptions);
  parser.parse(data);
  return parser.steal();
}
```
This does not need the mutex at all.
The mutex is only needed in the `catch` case, when setting `_errorMessage` and `_hasError`.

Additionally, the mutex needs to be acquired further down when testing/setting the `_hasError` and `_errorMessage` members.

The mutex we are holding here for a prolonged time is the per-`SenderThread` mutex. Each `SenderThread` is responsible for executing at most one request at a time. 
But there is the following code in ImportHelper.cpp that tries to find an idle `SenderThread` which may be impacted by the SenderThread holding its mutex for longer now.

```
/// Should return an idle sender, collect all errors
/// and return nullptr, if there was an error
SenderThread* ImportHelper::findIdleSender() {
  if (_autoUploadSize) {
    _autoTuneThread->paceSends();
  }

  while (!_senderThreads.empty()) {
    for (auto const& t : _senderThreads) {
      if (t->hasError()) {
        _hasError = true;
        _errorMessages.push_back(t->errorMessage());
        return nullptr;
      } else if (t->isIdle()) {
        return t.get();
      }    
    }    

    std::unique_lock guard{_threadsCondition.mutex};
    _threadsCondition.cv.wait_for(guard, std::chrono::milliseconds{10});
  }
  return nullptr;
}
```
The problem is that `SenderThread::hasError()` and `SenderThread::isIdle()` will both acquire the mutex as well, and holding the SenderThread's mutex for longer while parsing responses will impact the `ImportHelper::findIdleSender()` function for sure.
I can't tell if the impact is significant, but it could be the case.